lib/charms/tls_certificates_interface/v1/tls_certificates.py

```diff
@@ -1311,6 +1346,22 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
                         chain=certificate["chain"],
                     )
 
+    def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
+        """Handler triggered on relation broken event.
+
+        Args:
+            event: Juju event
+
+        Returns:
+            None
+        """
+        relation = self.model.get_relation(self.relationship_name)
```

Comment:
Using `self.relationship_name` here is probably fine if we have `limit: 1` in metadata but what about that `TooManyRelatedAppsError`?
I think idiomatically it should be `event.relation`.

Reply:
Was this resolved unintentionally?